core/libp2p/protocol_muxer/multiselect/multiselect_error.cpp

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/protocol_muxer/multiselect/multiselect_error.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(libp2p::protocol_muxer, MultiselectError, e) {
+  using Errors = libp2p::protocol_muxer::MultiselectError;
+  switch (e) {
+    case Errors::PROTOCOLS_LIST_EMPTY:
+      return "no protocols were provided";
+    case Errors::NEGOTIATION_FAILED:
+      return "there are no protocols, supported by both sides of the "
+             "connection";
+    case Errors::INTERNAL_ERROR:
+      return "internal error happened in this multiselect instance";
```

Comment:
Add another error, some kind of `PROTOCOL_VIOLATION` with a description that the other side has violated a protocol and sent something, not expected by us. This is to be used, for example, in `Multiselect::onUnexpectedRequestResponse`